### PR TITLE
add expression, payloadType and contentType to command model

### PR DESCRIPTION
**Pull request by AlvaroVega**

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,3 +1,4 @@
+Add: add expressions, payloadType and contentType to commands models
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]: No issue associated?

File: lib/templates/createDevice.json [hunk 2]
```diff
@@ -173,6 +173,18 @@
...
+                  "type": "string"
+                },
+                "payloadType": {
+                    "description": "Payload type",
+                    "type": "string"
+                },
+                "contentType": {
+                    "description": "Content type",
+                    "type": "string"
+                },
```
[the remark this answers is not in the document]
Comment (AlvaroVega) [reply lib/templates/createDevice.json:187]: expression, payloadType and contentType are optional fields of attributes:
https://github.com/telefonicaid/iotagent-node-lib/pull/1209/commits/2543f3993da986fd06a3e4fb43987044ad689e68

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (fgalan) [reply CHANGES_NEXT_RELEASE:1]: NTC

File: doc/api.md [hunk 3]
```diff
@@ -268,6 +268,14 @@ Some transformation plugins also allow the use of the following optional attribu
 -   **reverse**: add bidirectionality expressions to the attribute. See the **bidirectionality** transformation plugin
     in the [Data Mapping Plugins section](advanced-topics.md#bidirectionality-plugin-bidirectional) for details.
 
+Additionally for commands (which are attributes of `type` command) the following fields are optional:
```
Comment (fgalan) [inline doc/api.md:271]:
```suggestion
Additionally for commands (which are attributes of type `command`) the following fields are optional:
```

as `command` is the literal string to be used as type.

File: doc/api.md [hunk 4]
```diff
@@ -268,6 +268,14 @@ Some transformation plugins also allow the use of the following optional attribu
...
+
+-   **expression** indicates that the value of the target command will not be the plain value or the command, but an
+    expression based on a combination of the reported values. See the
```
Comment (fgalan) [inline doc/api.md:274]: Maybe better:

```suggestion
    expression based on a combination of the returned values. See the
```

File: doc/api.md [hunk 5]
```diff
@@ -268,6 +268,14 @@ Some transformation plugins also allow the use of the following optional attribu
...
+    [Expression Language definition](expressionLanguage.md) for details
+-   **payloadType**: indicates how command payload will be transformed before be sent to device
+-   **contentType**: `content-type` header used when send command by http transport
```
Comment (fgalan) [inline doc/api.md:277]:
```suggestion
-   **contentType**: `content-type` header used when send command by HTTP transport (ignored in other kinds of transports)
```

Unsure...

File: doc/api.md [hunk 6]
```diff
@@ -268,6 +268,14 @@ Some transformation plugins also allow the use of the following optional attribu
 -   **reverse**: add bidirectionality expressions to the attribute. See the **bidirectionality** transformation plugin
     in the [Data Mapping Plugins section](advanced-topics.md#bidirectionality-plugin-bidirectional) for details.
 
+Additionally for commands (which are attributes of `type` command) the following fields are optional:
+
+-   **expression** indicates that the value of the target command will not be the plain value or the command, but an
+    expression based on a combination of the reported values. See the
+    [Expression Language definition](expressionLanguage.md) for details
+-   **payloadType**: indicates how command payload will be transformed before be sent to device
```
Comment (fgalan) [inline doc/api.md:276]: Maybe the enumeration of supported values should be specified? Or, if this is not related with iotagent-node-lib, then a sentence like "Please have a look to particular IOTAs documentation for allowed values of this field in each case".

Comment (AlvaroVega) [reply doc/api.md:276]: I will comment "Please have a look to particular IOTAs documentation for allowed values of this field in each case" since is related to iotagent implementation

Comment (AlvaroVega) [reply doc/api.md]: Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1209/commits/51e40ec795236d39121aedf4bf5ee538a2a7aa16

**Review by fgalan — APPROVED** [review]
LGTM
